### PR TITLE
citra: 2020-12-07 -> 2021-11-01 & fix qt interface

### PR DESCRIPTION
**Pull request by superwhiskers**
###### motivation

citra was out of date and the qt ui didn't appear to work

###### things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSamus** [comment]
(I am not a maintainer)
You missed #142700 (by me) and #128541.
Since your version is newer you could incorporate the extra changes I made in the former.
The maintainers probably won't merge this in this state, since the package has too much vendoring (which is what they said in both PRs).
Don't ask me how to devendor though, I don't know,

**Review by jonringer — APPROVED** [review]
diff LGTM
gui launches fine

```
https://github.com/NixOS/nixpkgs/pull/144191

1 package built:
citra
```

**Comment by superwhiskers** [comment]
> (I am not a maintainer) You missed #142700 (by me) and #128541. Since your version is newer you could incorporate the extra changes I made in the former. The maintainers probably won't merge this in this state, since the package has too much vendoring (which is what they said in both PRs). Don't ask me how to devendor though, I don't know,

i apologize for not searching for existing prs (i was just trying to get a few changes i made locally to get the package running upstream), but it seems mine has been merged already.

**Pull request merged**
